File: TCDNew-master/src/ui.android/lib/models/dtos/PurchaseTaskLineUpdateDto.dart
import 'package:tiq/models/enums/PurchaseTaskLineUpdateType.dart';

import 'PurchaseTaskLineStateDto.dart';

class PurchaseTaskLineUpdateDto {
  String _purchaseTaskLineId;

  String purchaseTaskLineId() => _purchaseTaskLineId;
  PurchaseTaskLineUpdateType purchaseTaskLineUpdateType;
  PurchaseTaskLineStateDto purchaseTaskLineState;

  PurchaseTaskLineUpdateDto.empty(String purchaseTaskLineId) {
    this._purchaseTaskLineId = purchaseTaskLineId;
    this.purchaseTaskLineUpdateType = PurchaseTaskLineUpdateType.none;
    this.purchaseTaskLineState = null;
  }

  PurchaseTaskLineUpdateDto.withPurchaseTaskLineState(String purchaseTaskLineId, PurchaseTaskLineStateDto purchaseTaskLineState) {
    this._purchaseTaskLineId = purchaseTaskLineId;
    this.purchaseTaskLineUpdateType = PurchaseTaskLineUpdateType.none;
    this.purchaseTaskLineState = PurchaseTaskLineStateDto(
      purchaseTaskLineState.expirationDate,
      purchaseTaskLineState.expirationDaysPlus,
      0,
      0
    );
  }

  toJson() {
    return {
      'purchaseTaskLineId': _purchaseTaskLineId,
      'purchaseTaskLineUpdateType': purchaseTaskLineUpdateTypeToInt(purchaseTaskLineUpdateType),
      'purchaseTaskLineState': purchaseTaskLineState.toJson(),
    };
  }
}

File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Models/Dtos/PurchaseTaskLineUpdateDto.cs
using Cen.Wms.Client.Models.Enums;

namespace Cen.Wms.Client.Models.Dtos
{
    public class PurchaseTaskLineUpdateDto
    {
        private readonly string _purchaseTaskLineId;

        public string PurchaseTaskLineId { get { return _purchaseTaskLineId; } }
        public PurchaseTaskLineUpdateType PurchaseTaskLineUpdateType { get; set; }
        public PurchaseTaskLineStateDto PurchaseTaskLineState { get; set; }

        public PurchaseTaskLineUpdateDto(string purchaseTaskLineId)
        {
            _purchaseTaskLineId = purchaseTaskLineId;
            PurchaseTaskLineUpdateType = PurchaseTaskLineUpdateType.None;
            PurchaseTaskLineState = null;
        }

        public PurchaseTaskLineUpdateDto(string purchaseTaskLineId, PurchaseTaskLineStateDto purchaseTaskLineState)
        {
            _purchaseTaskLineId = purchaseTaskLineId;
            PurchaseTaskLineUpdateType = PurchaseTaskLineUpdateType.None;
            PurchaseTaskLineState = new PurchaseTaskLineStateDto
            {
                ExpirationDate = purchaseTaskLineState.ExpirationDate,
                ExpirationDaysPlus = purchaseTaskLineState.ExpirationDaysPlus,
                QtyNormal = 0,
                QtyBroken = 0
            };
        }
    }
}
